File: examples/java-junit-maven/src/test/java/examples/DeepThoughtSteps.java
package examples;

import dev.varar.Registrar;
import dev.varar.State;
import dev.varar.StateBinder;
import dev.varar.StepDefinitions;

public final class DeepThoughtSteps implements StepDefinitions {

    record Ctx() implements State {}

    @Override
    public void defineSteps(Registrar registrar) {
        StateBinder<Ctx> s = registrar.steps(Ctx::new);

        s.sensor("life, the universe and everything is {int}", (Ctx ctx, Integer answer) -> 42);
    }
}

File: examples/csharp-vstest/steps/deep-thought.steps.cs
using Varar;
using Varar.Core;

namespace Varar.Example;

public static class DeepThoughtSteps
{
    public static void Register(Steps s)
    {
        s.Sensor("life, the universe and everything is {int}", (state, answer) => Value.Of(42));
    }
}
